mobile_verifier/src/reward_shares.rs

```diff
@@ -197,6 +204,112 @@ impl MapperShares {
     }
 }
 
+#[derive(Default)]
+pub struct ServiceProviderShares {
+    pub shares: Vec<ServiceProviderDataSession>,
+}
+
+impl ServiceProviderShares {
+    pub fn new(shares: Vec<ServiceProviderDataSession>) -> Self {
+        Self { shares }
+    }
+
+    pub async fn from_payers_dc(
+        payer_shares: HashMap<String, u64>,
+        client: &dyn CarrierServiceVerifier<Error = ClientError>,
+    ) -> anyhow::Result<ServiceProviderShares> {
+        let mut shares = vec![];
```

Comment:
since `ServiceProviderShares` derives the default trait you can just `ServiceProviderShares::default()` here and then return that once you're done loading it up

Reply:
addressed here: 3f5924313ce8efe32a580e1981be7f4018355ee8